suse_migration_services/prechecks/kernels.py

```diff
@@ -0,0 +1,68 @@
+# Copyright (c) 2022
+#
+# SUSE Linux LLC.  All rights reserved.
+#
+# This file is part of suse-migration-services.
+#
+# suse-migration-services is free software: you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# suse-migration-services is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with suse-migration-services. If not, see <http://www.gnu.org/licenses/>
+#
+"""Checks the multiversion.kernels setting and the presence of multiple kernels
+
+Check to see if multiversion.kernels are enabled in /etc/zypp/zypp.conf and
+if there is more than one kernel currently installed. If detected, a warning is reported
+as multiversion.kernels needs to be disabled and old kernels cleaned up before
+a migration can be started."""
+
+import configparser
+import logging
+
+from suse_migration_services.defaults import Defaults
+from suse_migration_services.command import Command
+
+
+def multiversion_and_multiple_kernels():
+    """Check for the number of installed kernels"""
+    log = logging.getLogger(Defaults.get_migration_log_name())
+
+    config = configparser.ConfigParser()
+    config.read(Defaults.get_zypp_config_path())
+
+    current_settings = config.get('main', 'multiversion.kernels', fallback=None)
```

Comment:
This check is not sufficient. From zypp.conf

"""
Comma separated list of kernel packages to keep installed in parallel, if the
above multiversion variable is set.
"""

Meaning the `multiversion.kernels` setting only takes effect if `multiversion = provides:multiversion(kernel)` is also set. The `multiversion` variable could be set as shown (default), could be set to `multiversion=kernel`, could be set to something other than kernel or could not be set. We have to check the setting of `multiversion` first to avoid fals positive warnings.

Reply:
fixed ,please re-review